### PR TITLE
Adding types to Trove

### PR DESCRIPTION
**Pull request by prooheckcp**
Adds types to trove so that it becomes easier to work with typed luau while using Trove as the connection cleaner

### CODE REVIEW

File: modules/trove/init.lua [hunk 1]
```diff
@@ -160,7 +163,7 @@ end
 	end)
 	```
 ]=]
-function Trove:Connect(signal, fn)
+function Trove:Connect(signal, fn): any
```
[the remark this answers is not in the document]
Comment (Sleitnick) [reply modules/trove/init.lua:166]: RBXScriptSignal is unfortunately not a generic type

**Comment by Sleitnick** [comment]
I think it would be better to create a type of Trove that the constructor returns which contains all the typings. Also, we should avoid using the `any` type, as it is meaningless to the consumer.

**Comment by prooheckcp** [comment]
@Sleitnick I made some big changes to the typing, could you give some feedback about it?

**Comment by Sleitnick** [comment]
Typings added in latest Trove version

**Pull request closed without merging**